File: kai-printers-desktop/src-tauri/src/pos_dining_account_ticket.rs
//! Pre-cuenta / cuenta dining → JSON (`pos-dining-account-ticket`).

use crate::pos_sale_ticket_pdf::TicketCompany;
use anyhow::{Context, Result};
use serde::Deserialize;
use std::path::PathBuf;

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PosDiningAccountTicketLine {
    pub name: String,
    pub quantity: f64,
    pub unit_price: f64,
    pub line_total: f64,
    pub notes: Option<String>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PosDiningAccountTicketAccount {
    pub display_label: String,
    pub table_code: Option<String>,
    pub kind: String,
    #[allow(dead_code)]
    pub status: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PosDiningAccountTicketTotals {
    pub total: f64,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PosDiningAccountTicket {
    #[serde(default)]
    #[allow(dead_code)]
    pub version: i32,
    pub company: TicketCompany,
    pub account: PosDiningAccountTicketAccount,
    pub branch_name: Option<String>,
    pub point_of_sale_name: Option<String>,
    pub issued_at: String,
    pub lines: Vec<PosDiningAccountTicketLine>,
    pub totals: PosDiningAccountTicketTotals,
    pub footer_note: String,
}

pub fn parse_pos_dining_account_ticket_from_value(
    value: &serde_json::Value,
) -> Result<PosDiningAccountTicket> {
    serde_json::from_value(value.clone()).context("parse pos-dining-account-ticket")
}

pub fn write_pos_dining_account_ticket_escpos_from_value(
    dir: &PathBuf,
    value: &serde_json::Value,
) -> Result<PathBuf> {
    crate::pos_dining_account_ticket_escpos::write_pos_dining_account_ticket_escpos_from_value(
        dir, value,
    )
}

File: packages/print-service-client/src/pos-dining-account-ticket.ts
/**
 * Cuenta / pre-cuenta dining → agente KaiPrinters (`type: "pos-dining-account-ticket"`).
 * Documento informativo para el cliente (no es boleta ni ticket de venta).
 */

import type { PosSaleTicketCompany } from "./pos-sale-ticket";

export const POS_DINING_ACCOUNT_TICKET_PAYLOAD_VERSION = 1;

export const POS_DINING_ACCOUNT_TICKET_FOOTER_NOTE =
  "Documento informativo — no válido como boleta";

export type PosDiningAccountTicketLine = {
  name: string;
  quantity: number;
  unitPrice: number;
  lineTotal: number;
  notes?: string | null;
};

export type PosDiningAccountTicketAccount = {
  displayLabel: string;
  tableCode?: string | null;
  kind: string;
  status: string;
};

export type PosDiningAccountTicketPayload = {
  version: number;
  company: PosSaleTicketCompany;
  account: PosDiningAccountTicketAccount;
  branchName?: string | null;
  pointOfSaleName?: string | null;
  issuedAt: string;
  lines: PosDiningAccountTicketLine[];
  totals: {
    total: number;
    /** Propina sugerida (informativa; no fiscal). */
    tipSuggestedAmount?: number | null;
    tipSuggestPercent?: number | null;
    totalWithTip?: number | null;
  };
  footerNote: string;
};

export type PosDiningAccountTicketPrintExtras = {
  filename: string;
  documentType?: string;
  internalFolio?: string;
  sourceApp?: string;
  purpose?: string;
};
